web-common/src/features/connectors/selectors.ts

```diff
@@ -56,15 +68,21 @@ function createModelingSupportQueryOptions(
       const spec = data?.resource?.connector?.spec;
       if (!spec) return false;
 
-      // Modeling is supported if:
-      // - DuckDB (embedded database with full SQL support)
-      // - Provisioned (managed) connectors
-      // - Read-write mode connectors
-      return (
-        spec.driver === "duckdb" ||
-        spec.provision === true ||
-        spec.properties?.mode === "readwrite"
-      );
+      if (modelingType === "sql") {
+        // SQL modeling is supported if:
+        // - DuckDB (embedded database with full SQL support)
+        // - Provisioned (managed) connectors
+        // - Read-write mode connectors
+        return isSqlModelingSupported(spec);
+      } else {
+        // YAML modeling is supported for connectors that support connector type specification:
+        // These connectors use YAML files with "connector: postgres" syntax
+        // - Postgres, MySQL, SQL Server, etc. (external databases)
+        // - NOT DuckDB (uses SQL models)
+        // - NOT provisioned connectors (uses SQL models)
+        // - NOT read-write mode connectors (uses SQL models)
+        return !isSqlModelingSupported(spec);
+      }
```

Comment:
This is confusing because, when modeling is supported, the user can use either SQL or YAML – whatever they prefer. (However, we're opinionated and think that SQL should be used for basic models and YAML should be used for [advanced models](https://docs.rilldata.com/build/advanced-models/).)

So, to avoid confusion, it feels like function and variable names need to be changed.

Reply:
https://github.com/rilldata/rill/pull/7716#discussion_r2255144647